webrender/res/ps_border_edge.vs.glsl

```diff
@@ -7,52 +7,97 @@ void write_edge_distance(float p0,
                          float original_width,
                          float adjusted_width,
                          float style,
-                         float axis_select) {
+                         float axis_select,
+                         float sign_adjust) {
     switch (int(style)) {
-        case BORDER_STYLE_DOUBLE: {
+        case BORDER_STYLE_DOUBLE:
             vEdgeDistance = vec2(p0 + adjusted_width,
                                  p0 + original_width - adjusted_width);
             break;
-        }
-        default: {
+        case BORDER_STYLE_GROOVE:
+        case BORDER_STYLE_RIDGE:
+            vEdgeDistance = vec2(p0 + adjusted_width, sign_adjust);
+            break;
+        default:
             vEdgeDistance = vec2(0.0);
             break;
-        }
     }
 
     vAxisSelect = axis_select;
 }
 
+void write_alpha_select(float style) {
+    switch (int(style)) {
+        case BORDER_STYLE_DOUBLE:
+            vAlphaSelect = 0.0;
+            break;
+        default:
+            vAlphaSelect = 1.0;
+            break;
+    }
+}
+
+void write_color(vec4 color, float style, bool flip) {
+    vec2 modulate;
+    if (flip) {
+        modulate = vec2(0.7, 1.3);
+    } else {
+        modulate = vec2(1.3, 0.7);
+    }
+
+    switch (int(style)) {
+        case BORDER_STYLE_GROOVE:
+            vColor0 = vec4(color.rgb * modulate.y, color.a);
```

Comment:
similarly, could be simplified by leaving only the `modulate` computation in the match